### PR TITLE
Added User Role Check endpoint, fixed WOB-218

### PR DESCRIPTION
**Pull request by Assstarte**
Fixed

### CODE REVIEW

File: project/application/src/pages/TeamPage/index.js [hunk 1]
```diff
@@ -192,6 +197,9 @@ class TeamPage extends Component {
         //@TODO Get Saved value from localStorage
         fetch(AppConfig.apiURL + `team/current/?userId=${getUserIdFromLocalStorage()}`).then(res => {
```
Comment (athlonUA) [inline project/application/src/pages/TeamPage/index.js]: remove, get current team data from localStorage

File: project/application/src/components/EditTeamModal/index.js [hunk 2]
```diff
@@ -66,8 +67,21 @@ class EditTeamModal extends Component {
             );
     };
 
+    checkRole() {
```
Comment (athlonUA) [inline project/application/src/components/EditTeamModal/index.js]: remove, get role from current user you want to edit

File: project/application/src/services/userStorageService.js [hunk 3]
```diff
@@ -11,8 +13,8 @@ export function getUserEmailFromLocalStorage() {
 }
 
 export function getUserRoleTitleFromLocalStorage() {
-    const title = (getUserFromLocalStorage().roleCollaboration || {}).title || '';
-
+    const accessData = getCurrentTeamAccessLevelFromLocalStorage();
```
Comment (athlonUA) [inline project/application/src/services/userStorageService.js]: get role from current team obj (localStorage)

File: project/application/src/components/EditTeamModal/index.js [hunk 4]
```diff
@@ -18,6 +18,7 @@ class EditTeamModal extends Component {
         id: null,
         value: ROLES.ROLE_MEMBER,
         valueStatus: USER_STATUS.NOT_ACTIVE,
+        currentRole: ROLES.ROLE_MEMBER,
```
Comment (athlonUA) [inline project/application/src/components/EditTeamModal/index.js]: The `currentRole` duplicates `value`. Remove `currentRole`

File: project/application/src/components/EditTeamModal/index.js [hunk 5]
```diff
@@ -67,7 +68,7 @@ class EditTeamModal extends Component {
     };
 
     handleChange = event => {
-        this.setState({ value: event.target.value });
+        this.setState({ currentRole: event.target.value });
```
Comment (athlonUA) [inline project/application/src/components/EditTeamModal/index.js]: Revert back

File: project/application/src/components/EditTeamModal/index.js [hunk 6]
```diff
@@ -101,7 +102,7 @@ class EditTeamModal extends Component {
                     </div>
                     <div className="edit_team_modal_input_container">
                         <div className="edit_team_modal_input_title">Role</div>
-                        <RadioGroup onChange={this.handleChange} value={this.state.value}>
+                        <RadioGroup onChange={this.handleChange} value={this.state.currentRole}>
```
Comment (athlonUA) [inline project/application/src/components/EditTeamModal/index.js]: Revert back

File: project/application/src/components/EditTeamModal/index.js [hunk 7]
```diff
@@ -136,7 +137,10 @@ class EditTeamModal extends Component {
     }
 
     componentDidMount() {
-        this.setState({ id: this.props.editedUser.id });
+        this.setState({
+            id: this.props.editedUser.id,
+            currentRole: this.props.editedUser.role,
```
Comment (athlonUA) [inline project/application/src/components/EditTeamModal/index.js]: Replace `currentRole` with `value`

File: project/application/src/components/TeamSwitcher/index.js [hunk 8]
```diff
@@ -45,7 +46,21 @@ class TeamSwitcher extends Component {
...
         }).then(res =>
             res.json().then(response => {
-                window.location.reload(true);
+                fetch(AppConfig.apiURL + `team/current/?userId=${getUserIdFromLocalStorage()}`).then(res =>
+                    res.json().then(response => {
+                        this.setState({
+                            currentTeamId: response.data.user_team[0].team.id,
+                            currentTeamName: response.data.user_team[0].team.name,
+                        });
+                        console.log(response);
```
Comment (athlonUA) [inline project/application/src/components/TeamSwitcher/index.js]: Remove cl

File: project/application/src/components/TeamSwitcher/index.js [hunk 9]
```diff
@@ -45,7 +46,21 @@ class TeamSwitcher extends Component {
             }),
         }).then(res =>
             res.json().then(response => {
-                window.location.reload(true);
+                fetch(AppConfig.apiURL + `team/current/?userId=${getUserIdFromLocalStorage()}`).then(res =>
+                    res.json().then(response => {
+                        this.setState({
+                            currentTeamId: response.data.user_team[0].team.id,
```
Comment (athlonUA) [inline project/application/src/components/TeamSwitcher/index.js]: I see that now we send a request to the server to change current team & update info on the front. So we need to remove lines from 27 to 36

File: project/application/src/pages/TeamPage/index.js [hunk 10]
```diff
@@ -84,26 +89,32 @@ class TeamPage extends Component {
...
-                <td>
-                    <div>{element.user[0].is_active ? 'Active' : 'Not active'}</div>
-                    {checkIsAdmin() && (
-                        <i onClick={e => this.openEditMiodal(element.user[0])} className="edit_button item_button" />
-                    )}
-                </td>
-            </tr>
-        ));
+        const items = programersArr.map((element, index) => {
+            element.user[0].role = element.role_collaboration.title;
```
Comment (athlonUA) [inline project/application/src/pages/TeamPage/index.js]: This assignment is not used anywhere in this scope. Revert back the changes in this block

File: project/application/src/pages/TeamPage/index.js [hunk 11]
```diff
@@ -190,13 +202,11 @@ class TeamPage extends Component {
...
         //@TODO Get Saved value from localStorage
-        fetch(AppConfig.apiURL + `team/current/?userId=${getUserIdFromLocalStorage()}`).then(res => {
-            res.json().then(response => {
-                this.setState({
-                    teamName: response.data.user_team[0].team.name,
-                    teamId: response.data.user_team[0].team.id,
-                });
-            });
+        let teamData = getCurrentTeamDataFromLocalStorage();
+        console.log(teamData);
```
Comment (athlonUA) [inline project/application/src/pages/TeamPage/index.js]: remove cl

File: project/application/src/services/userStorageService.js [hunk 12]
```diff
@@ -11,9 +13,9 @@ export function getUserEmailFromLocalStorage() {
 }
 
 export function getUserRoleTitleFromLocalStorage() {
```
Comment (athlonUA) [inline project/application/src/services/userStorageService.js]: Replace with this

```
export function getUserRoleTitleFromLocalStorage() {
    const role = getCurrentTeamDataFromLocalStorage().role || '';

    return role;
}

File: project/application/src/pages/TeamPage/index.js [hunk 13]
```diff
@@ -11,6 +11,11 @@ import { checkIsAdminByRole, checkIsMemberByRole, userLoggedIn, checkIsAdmin } f
 import EditTeamModal from '../../components/EditTeamModal';
 import { AppConfig } from '../../config';
 import { getUserIdFromLocalStorage } from '../../services/userStorageService';
+import {
+    setCurrentTeamDataToLocalStorage,
+    setCurrentTeamAccessLevelToLocalStorage,
+    getCurrentTeamDataFromLocalStorage,
+} from '../../services/teamStorageService';
```
Comment (athlonUA) [inline project/application/src/pages/TeamPage/index.js]: I think we can remove all these imports as its not used anywhere

File: project/application/src/pages/TeamPage/index.js (hunk 10 above)
Comment (Assstarte) [reply project/application/src/pages/TeamPage/index.js]: We still need it for the role check.

File: project/application/src/services/userStorageService.js (hunk 12 above)
Comment (athlonUA) [reply project/application/src/services/userStorageService.js]: ok

**Pull request merged**